code/game/objects/items/tanks/tanks.dm

```diff
@@ -281,3 +281,27 @@
 
 	else if(integrity < 3)
 		integrity++
+
+/obj/item/tank/rad_act(strength)
+	. = ..()
+	var/gas_change = FALSE
+	var/list/cached_gases = air_contents.gases
+	if(cached_gases[/datum/gas/oxygen] && cached_gases[/datum/gas/carbon_dioxide])
+		gas_change = TRUE
+		var/pulse_strength = min(strength, cached_gases[/datum/gas/oxygen][MOLES] * 1000, cached_gases[/datum/gas/carbon_dioxide][MOLES] * 2000)
+		cached_gases[/datum/gas/carbon_dioxide][MOLES] -= pulse_strength / 2000
+		cached_gases[/datum/gas/oxygen][MOLES] -= pulse_strength / 1000
+		ASSERT_GAS(/datum/gas/pluoxium, air_contents)
+		cached_gases[/datum/gas/pluoxium][MOLES] += pulse_strength / 4000
+		strength -= pulse_strength
+
+	if(cached_gases[/datum/gas/hydrogen])
+		gas_change = TRUE
+		var/pulse_strength = min(strength, cached_gases[/datum/gas/hydrogen][MOLES] * 1000)
+		cached_gases[/datum/gas/hydrogen] -= pulse_strength / 1000
+		ASSERT_GAS(/datum/gas/tritium, air_contents)
+		cached_gases[/datum/gas/tritium][MOLES] += pulse_strength / 1000
+		strength -= pulse_strength
+
+	if(gas_change)
+		air_contents.garbage_collect()
```

Comment:
In a perfect world this would be one proc that you pass gasmixtures into, but I'm also not sure how I feel about putting rad_act() on the gasmixture datum, so perhaps not

Reply:
Same honestly.

Reply:
I mean. I could absolutely put something like that on gas mixtures. It might even help in some cases, because immutable mixtures can just tell it to not. It would solve the uranium doors on the gluttony ruin anyway.

Reply:
immutable mixes would get caught by the garbage_collect, might make things more cluttered, specially since planetary mixes aren't any different then normal, they're denoted by a bool on turfs, not by anything in the mix itself